### ISSUE TITLE
Page numbers and transcriber credit

### ISSUE THREAD

**Issue by DanielHeidt** [TheCanadianConfederationDebates/TCCD#37]
As of a few weeks ago, the Conestoga transcription platform has separate fields for page numbers (as shown in top-right corner of image below).
<img width="2021" alt="transcribing" src="https://cloud.githubusercontent.com/assets/16122501/20688319/5aeedc8a-b58e-11e6-8497-f88b4b0583e9.png">

There are, as you can see, two fields. Most pages will only have one number, but the Alb-Sask instead have column numbers. Since no separate page number exists, these are used by all scholars as page numbers.

Since these have to be manually entered by transcribers, they are more reliable than the OCR results. They also always appear in the same place, so they should be very easy for a script to find and utilize.

From Martin's description, the transcription platform enters three bits of info:
1) a first page number (which should always have a value)
2) a second page number (which will be null except when there are two column numbers on the page)
3) the transcriber's email address (harvested at login). These should be harvested at a later date to populate the "who transcribed this debate day" as we previously discussed in another meeting.

**Comment by martindholmes** [comment]
Assigning this back to Dan, because I think page numbers are working fine at this point, and on the other issue:

I've examined a large number of the transcribed/corrected pages retrieved from the transcription platform, and I can't find any sign that the transcriber's information is included. For example: 

https://github.com/TheCanadianConfederationDebates/TCCD/blob/master/data/PEI/Provincial/hocr_edited/PEI_LA_Page_018001-edit275.html

Can you see any info about the transcriber/proofer there? If it's not there, I can't harvest it for inclusion in the TEI metadata.

**Comment by DanielHeidt** [comment]
Please refer to the email I have re-forwarded you from 24 Oct 2016. Only pages transcribed from roughly this date, and after, will have the tag. The transcriber info is encoded, for example, in:

https://github.com/TheCanadianConfederationDebates/TCCD/blob/master/data/Nfld/Provincial/National_Convention/hocr_edited/NFLD_National_Convention_Page_40764-edit1757.html

The page numbers, however, do not seem to be there consistently. I think they are there when a transcriber entered them, but the script might not be firing when a reviewer enters them. I will pass this observation on to the coders. In the meantime, I am passing this back to you as the transcriber/proofer is being encoded correctly.

**Comment by martindholmes** [comment]
Next question, then: what are we to do with this information? We can't blithely publish transcribers' email addresses without their permission; that's a violation of privacy, presumably. Nor can we infer someone's name from their email address.

I can easily harvest this into the TEI, of course, but it's more difficult to figure out what to do with it in the website output.

**Comment by DanielHeidt** [comment]
I assumed we would harvest the usernames only (discarding everything after "@"). Does that sound ok?

**Comment by martindholmes** [comment]
That's a sort of privacy-by-obscurity. And now I think about it, of course we can't put them in the XML because that ends up public on GitHub. :-) We could encrypt them with a key we keep out of the repo as a temporary measure, but it's a bit messy.

We'll need to contact users to get their permission to share their name or their email address or both. That should be included in the sign-up form on the transcription platform, but for existing users we'll probably have to contact them.

**Comment by DanielHeidt** [comment]
That is, of course, pretty involved.... Aside from the email, it would require significant changes to the system to distinguish between those who want to be tracked and those who don't. Wouldn't it be simpler to temporarily make the repo private instead of doing all of that extra coding? We can erase the emails before making them public again. As you'll see on the achievement charts at http://theconfederationdebates.ca/, we have already been using usernames for months without any complaint. So long as we're not publishing the email domains, I don't see the problem.

**Comment by martindholmes** [comment]
Taking this discussion to email for a bit...

**Comment by martindholmes** [comment]
I'm not sure where we're at with this. All the HOCR data I have contains something that looks like this:

`Corrected by [Martin Holmes] on [the date, as
2016-02-22].`

In other words, full names, not usernames or email addresses. This means that full names are already in the public git repo. I can write some code to transfer them into the equivalent XML pages, and then into the HTML; is that appropriate?

**Comment by DanielHeidt** [comment]
I think your example is from the old Dropbox workflow. Files from the transcriber portal also had an email address or username encoded (ex. <meta name="email" content="jesshobbit41"> in NS_LA_Page_026309-edit1840.html). 

To implement this feature fully, we need to harvest this data and post it to each debate day. It appears that this data is in each of the hocr files in the "final" directory. I assume some sort of comparison of a string from each debate day page with the final debate day xml file is still required. I am glad to chat about this at our next meeting.

**Comment by martindholmes** [comment]
Right. There are about 3,200 instances of those usernames. Many of those documents haven't yet been turned into XML, though, so it doesn't really make sense to try to script the process now. Every time we finish all the pages from a specific province, we should do this, I guess. Are any provinces completely done?

**Comment by martindholmes** [comment]
Per DH: Morris is completely done; BC is completely done (there is no French). The current idea is this:

1. Load the XML file for a debate day.
2. For each source HOCR document, load the document and find either p[@class='editorial'] and get the name after "Corrected by" (for old files), or //meta[@name='user']/@content (make sure to get unique values only, since they're duplicated many times in the same file).

**Comment by DanielHeidt** [comment]
Please insert the credits between "Source" and "Participating individuals" near the bottom of each debate day.

**Comment by martindholmes** [comment]
XML files for BC now have proofing attribution. Nothing in the output yet.

**Comment by martindholmes** [comment]
Output should now be there, but waiting on fix for team.html before I can build the site.

**Comment by martindholmes** [comment]
This is now working. The only remaining thing is to provide the French caption (it says "Credits" for both English and French).

**Comment by DanielHeidt** [comment]
I'm finally back into the project. Since all of the pages are now transcribed, is there any chance we can script the credits for the rest of the provinces? Not sure if your script currently captures the older dropbox style names where users inputted their names next to "corrected by" in the green header you created? If yes, this should capture the contributions of nearly all participants. Any chance you could try to apply your script to the rest of the legislatures sometime in the new few weeks?

**Comment by martindholmes** [comment]
It'll be really hard to find any time for this project in the next few weeks, but I'll try. HCMC's original commitment of time has long expired, unfortunately, and we have a huge amount of work on right now.

**Comment by DanielHeidt** [comment]
So far I know, team.html is working - I made a minor typo (which I believe is fixed) that was hampering your build at the time.

**Comment by martindholmes** [comment]
It's working, but it's ugly. I hacked it: 

https://github.com/TheCanadianConfederationDebates/TCCD/issues/110

So we can ignore that one for now.
